packages/@aws-cdk/aws-fsx/lib/lustre-file-system.ts

```diff
@@ -0,0 +1,305 @@
+
+  /**
+   * The security groups/rules used to allow network connections to the file system.
+   */
+  public readonly connections: Connections;
+
+  /**
+   * The ID that AWS assigns to the file system.
+   */
+  public readonly fileSystemId: string;
+
+  /**
+   * The encapsulated L1 file system.
+   */
+  private readonly fsxFileSystem: CfnFileSystem;
+
+  constructor(scope: Construct, id: string, props: FsxLustreFileSystemProps) {
+    super(scope, id);
+
+    this.validateProps(props);
+
+    const securityGroup = (props.securityGroup || new SecurityGroup(this, 'FsxLustreSecurityGroup', {
+      vpc: props.vpc
+    }));
+    securityGroup.addIngressRule(Peer.anyIpv4(), Port.tcp(FsxLustreFileSystem.DEFAULT_PORT), 'Default FSx port ingress');
+    securityGroup.addEgressRule(Peer.anyIpv4(), Port.tcp(FsxLustreFileSystem.DEFAULT_PORT), 'Default FSx port egress');
+
+    this.connections = new Connections({
+      securityGroups: Array.of(securityGroup),
+      defaultPort: Port.tcp(FsxLustreFileSystem.DEFAULT_PORT)
+    });
+
+    this.fsxFileSystem = new CfnFileSystem(this, 'FsxLustreFileSystem', {
+      fileSystemType: FsxLustreFileSystem.DEFAULT_FILE_SYSTEM_TYPE,
+      subnetIds: Array.of(props.vpcSubnet.subnetId),
+      backupId: props.backupId,
+      kmsKeyId: (props.kmsKey ? props.kmsKey.keyId : undefined),
+      lustreConfiguration: props.lustreConfiguration,
+      securityGroupIds: Array.of(securityGroup.securityGroupId),
+      storageCapacity: props.storageCapacity
+    });
+
+    this.fileSystemId = this.fsxFileSystem.ref;
+    this.node.defaultChild = this.fsxFileSystem;
+  }
+
+  /**
+   * Validates the props provided for a new FSx for Lustre file system.
+   */
+  private validateProps(props: FsxLustreFileSystemProps) {
+    const lustreConfiguration = props.lustreConfiguration;
+    const deploymentType = lustreConfiguration.deploymentType;
+
+    this.validateExportPath(lustreConfiguration.exportPath, lustreConfiguration.importPath);
+    this.validateImportedFileChunkSize(lustreConfiguration.importedFileChunkSize);
+    this.validateImportPath(lustreConfiguration.importPath);
+    this.validatePerUnitStorageThroughput(deploymentType, lustreConfiguration.perUnitStorageThroughput);
+    this.validateStorageCapacity(deploymentType, props.storageCapacity);
+    this.validateWeeklyMaintenanceStartTime(lustreConfiguration.weeklyMaintenanceStartTime);
+  }
+
+  /**
+   * Validates the export path is in the correct format and matches the import path.
+   */
+  private validateExportPath(exportPath?: string, importPath?: string): void {
+    if (exportPath === undefined) { return; }
+
+    if (importPath === undefined) {
+      throw new Error('Cannot define an export path without also defining an import path');
+    }
+    if (!exportPath.startsWith(importPath)) {
+      throw new Error(`The export path "${exportPath}" is invalid. Expecting the format: s3://{IMPORT_PATH}/optional-prefix`);
```

Comment:
Do both need to start with `s3://` ? Might be useful to validate as well if so.

Reply:
Oh haha, I see this is implicitly checked by validation of import path.

Downside is that this is called before `validateImportPath()`, so first you'll get an error that tells you you need to add importPath and only THEN will it tell you you need to prefix `s3://`.

Reply:
Oh good catch, I had them validating alphabetically and didn't think of that. It makes sense to do the import path validation first. I'll change the ordering.